core/common/src/annotation/TextBlock.ts

```diff
@@ -402,6 +406,146 @@ export class TabRun extends TextBlockComponent {
   }
 }
 
+/** An entry in a [[FieldPropertyPath]].
+ * @beta
+ */
+export interface FieldPropertyPathEntry {
+  /** The name of the property. */
+  propertyName: string;
+  /** If [[propertyName]] identifies an array property, the index of the array element of interest.
+   * Negative array indices can be used to count backward from the end of the array (e.g., index -1 refers to the last element, -2 to the second-to-last element, and so on).
+   */
+  arrayIndex?: number;
+}
+
+/** A chain of property accesses that resolves to a primitive value that forms the basis of the displayed content
+ * of a [[FieldRun]].
+ * The chain may traverse through structs, arrays, and JSON objects.
+ * @beta
+ */
+export type FieldPropertyPath = FieldPropertyPathEntry[];
```

Comment:
@khanaffan @grigasp is there an existing/better way to encode an arbitrarily complex access string that may traverse arrays, structs, and JSON properties?

Reply:
IN ECSql you can access using SQLite [JSON1](https://www.sqlite.org/changes.html) functions

```
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$') → '{"a":2,"c":[4,5,{"f":7}]}'
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$.c') → '[4,5,{"f":7}]'
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$.c[2]') → '{"f":7}'
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$.c[2].f') → 7
SELECT json_extract('{"a":2,"c":[4,5],"f":7}','$.c','$.a') → '[[4,5],2]'
SELECT json_extract('{"a":2,"c":[4,5],"f":7}','$.c[#-1]') → 5
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$.x') → NULL
SELECT json_extract('{"a":2,"c":[4,5,{"f":7}]}', '$.x', '$.a') → '[null,2]'
SELECT json_extract('{"a":"xyz"}', '$.a') → 'xyz'
SELECT json_extract('{"a":null}', '$.a') → NULL
```

Reply:
I should add that while my main goal is to fetch the property value, I'll almost certainly need some metadata about the property, like its kind-of-quantity and extended type.